dask/dataframe/io/parquet/utils.py

```diff
@@ -476,7 +477,12 @@ def _scrub(i, p):
             "/".join(path_parts[l:])
         )  # use '/'.join() instead of _join_path to be consistent with split('/')
 
+    # numeric rather than glob ordering
+    out_list = sorted(out_list, key=natural_sort_key)
+    file_list = sorted(file_list, key=natural_sort_key)
+
     return (
+        file_list,
         "/".join(basepath),
         out_list,
     )  # use '/'.join() instead of _join_path to be consistent with split('/')
```

Comment:
I realize that `_analyze_paths` is supposed to be "private",  but there are other libraries (like [nvtabular](https://github.com/NVIDIA/NVTabular/blob/b1c842571fc8edb92cc048b98293b56b2088edcb/nvtabular/io/parquet.py#L239)) that are actually using this utility (and expecting two elements).  It is probably best for those libraries to move away from using any private APIs asap, but I'd like to avoid changing the API (at leasts for default usage) in the short term.

Perhaps you could either (1) Introduce an optional `sort=` kwarg that will result in the sorted paths being returned, or (2) Introduce a simple `_sort_and_analyze_paths` utility to wrap the existing `_analyze_paths` (sorting input paths before, and out_list after).  Sorry - I know your current solution is a bit cleaner :/

I also agree with @jsignell that it may be best to sort the input paths at the begining of (or before) `_analyze_paths`.

Reply:
I simplified things by just sorting at every `_analyze_paths` call-site, prior to `_analyze_paths`.

Re-reading your comment now, I guess that wasn't actually one of your suggested fixes 😁

Making a `_sort_and_analyze_paths` and using it in this code-base (and encouraging any ecosystem users to switch to it) sounds good. I guess a `sort=False` param to `_analyze_paths` that changes the arity of the returned tuple works too.

Let me know what you prefer.

Reply:
alright I went ahead and moved to `_sort_and_analyze_paths`; I'll push it here once CI passes on the current commit, so we have two ✅s to choose from